File: nextjs-backend/prisma/seed.ts
import { PrismaClient } from '@prisma/client';
import * as bcrypt from 'bcrypt';

const prisma = new PrismaClient();

async function main() {
  console.log('🌱 Starting database seed...');

  // Clear existing data
  await prisma.auditLog.deleteMany();
  await prisma.directMessage.deleteMany();
  await prisma.directConversationParticipant.deleteMany();
  await prisma.directConversation.deleteMany();
  await prisma.feedbackComment.deleteMany();
  await prisma.file.deleteMany();
  await prisma.notification.deleteMany();
  await prisma.announcement.deleteMany();
  await prisma.feedback.deleteMany();
  await prisma.channelMember.deleteMany();
  await prisma.channel.deleteMany();
  await prisma.user.deleteMany();

  console.log('✅ Cleared existing data');

  // Hash passwords
  const demoPassword = await bcrypt.hash('P@ssw0rd!', 10);
  const defaultPassword = await bcrypt.hash('password123', 10);

  // Create demo users (new credentials)
  const superAdmin = await prisma.user.create({
    data: {
      email: 'sa@demo.local',
      password: demoPassword,
      fullName: 'Super Administrator',
      role: 'SUPERADMIN',
      department: 'Executive',
      dateOfBirth: new Date('1980-01-15'),
      employeeId: 'EMP-SUP-001',
      isActive: true,
    },
  });

  const demoAdmin = await prisma.user.create({
    data: {
      email: 'admin@demo.local',
      password: demoPassword,
      fullName: 'Demo Admin',
      role: 'ADMIN',
      department: 'IT',
      dateOfBirth: new Date('1985-03-03'),
      employeeId: 'EMP-ADM-001',
      isActive: true,
    },
  });

  const demoUser = await prisma.user.create({
    data: {
      email: 'user@demo.local',
      password: demoPassword,
      fullName: 'Demo Employee',
      role: 'EMPLOYEE',
      department: 'Sales',
      dateOfBirth: new Date('1990-05-20'),
      employeeId: 'EMP-EMP-001',
      isActive: true,
    },
  });

  // Create existing users (keep backward compatibility)
  const admin = await prisma.user.create({
    data: {
      email: 'admin@company.com',
      password: defaultPassword,
      fullName: 'Admin User',
      role: 'ADMIN',
      department: 'IT',
      dateOfBirth: new Date('1988-02-10'),
      employeeId: 'EMP-ADMIN-001',
      isActive: true,
    },
  });

  const hrManager = await prisma.user.create({
    data: {
      email: 'hr@company.com',
      password: defaultPassword,
      fullName: 'HR Manager',
      role: 'HR',
      department: 'Human Resources',
      dateOfBirth: new Date('1987-04-05'),
      employeeId: 'EMP-HR-001',
      isActive: true,
    },
  });

  const employee1 = await prisma.user.create({
    data: {
      email: 'john.doe@company.com',
      password: defaultPassword,
      fullName: 'John Doe',
      role: 'EMPLOYEE',
      department: 'Engineering',
      dateOfBirth: new Date('1992-08-12'),
      employeeId: 'EMP-ENG-001',
      isActive: true,
    },
  });

  const employee2 = await prisma.user.create({
    data: {
      email: 'jane.smith@company.com',
      password: defaultPassword,
      fullName: 'Jane Smith',
      role: 'EMPLOYEE',
      department: 'Marketing',
      dateOfBirth: new Date('1991-09-23'),
      employeeId: 'EMP-MKT-001',
      isActive: true,
    },
  });

  const employee3 = await prisma.user.create({
    data: {
      email: 'bob.johnson@company.com',
      password: defaultPassword,
      fullName: 'Bob Johnson',
      role: 'EMPLOYEE',
      department: 'Sales',
      dateOfBirth: new Date('1993-11-30'),
      employeeId: 'EMP-SLS-001',
      isActive: true,
    },
  });

  const employee4 = await prisma.user.create({
    data: {
      email: 'alice.williams@company.com',
      password: defaultPassword,
      fullName: 'Alice Williams',
      role: 'EMPLOYEE',
      department: 'Engineering',
      dateOfBirth: new Date('1994-12-08'),
      employeeId: 'EMP-ENG-002',
      isActive: true,
    },
  });

  console.log('✅ Created 7 users');

  // Create channels
  const generalChannel = await prisma.channel.create({
    data: {
      name: 'General Announcements',
      description: 'Company-wide announcements and updates',
      channelType: 'announcement',
      isPrivate: false,
      createdBy: admin.id,
    },
  });

  const hrChannel = await prisma.channel.create({
    data: {
      name: 'HR Updates',
      description: 'Human Resources policies and updates',
      channelType: 'hr',
      isPrivate: false,
      createdBy: hrManager.id,
    },
  });

  const engChannel = await prisma.channel.create({
    data: {
      name: 'Engineering Team',
      description: 'Engineering team discussions',
      channelType: 'team',
      isPrivate: true,
      createdBy: employee1.id,
    },
  });

  console.log('✅ Created 3 channels');

  // Create channel memberships
  await prisma.channelMember.createMany({
    data: [
      { userId: admin.id, channelId: generalChannel.id, role: 'MODERATOR' },
      { userId: hrManager.id, channelId: generalChannel.id, role: 'MEMBER' },
      { userId: employee1.id, channelId: generalChannel.id, role: 'MEMBER' },
      { userId: employee2.id, channelId: generalChannel.id, role: 'MEMBER' },
      { userId: hrManager.id, channelId: hrChannel.id, role: 'MODERATOR' },
      { userId: employee1.id, channelId: hrChannel.id, role: 'MEMBER' },
      { userId: employee2.id, channelId: hrChannel.id, role: 'MEMBER' },
      { userId: employee1.id, channelId: engChannel.id, role: 'MODERATOR' },
      { userId: employee4.id, channelId: engChannel.id, role: 'MEMBER' },
    ],
  });

  console.log('✅ Created channel memberships');

  // Create feedback
  const feedback1 = await prisma.feedback.create({
    data: {
      title: 'Improve office ergonomics',
      description: 'The current desk chairs are causing back pain. We need better ergonomic furniture.',
      category: 'WORKPLACE',
      status: 'SUBMITTED',
      isAnonymous: false,
      submittedBy: employee1.id,
    },
  });

  const feedback2 = await prisma.feedback.create({
    data: {
      title: 'Request for remote work policy',
      description: 'Would like to see a more flexible remote work policy implemented.',
      category: 'MANAGEMENT',
      status: 'UNDER_REVIEW',
      isAnonymous: false,
      submittedBy: employee2.id,
      assignedTo: hrManager.id,
    },
  });

  const feedback3 = await prisma.feedback.create({
    data: {
      title: 'Health insurance concerns',
      description: 'The current health insurance plan has limited coverage. Would like better options.',
      category: 'BENEFITS',
      status: 'IN_PROGRESS',
      isAnonymous: true,
      submittedBy: employee3.id,
      assignedTo: hrManager.id,
    },
  });

  const feedback4 = await prisma.feedback.create({
    data: {
      title: 'Positive team culture',
      description: 'Really enjoying the collaborative culture in the engineering team!',
      category: 'CULTURE',
      status: 'RESOLVED',
      isAnonymous: false,
      submittedBy: employee4.id,
      assignedTo: hrManager.id,
    },
  });

  console.log('✅ Created 4 feedback items');

  // Create feedback comments
  await prisma.feedbackComment.createMany({
    data: [
      {
        feedbackId: feedback2.id,
        userId: hrManager.id,
        comment: 'Thank you for your feedback. We are reviewing remote work options.',
        isInternal: false,
      },
      {
        feedbackId: feedback3.id,
        userId: hrManager.id,
        comment: 'Working with insurance providers to improve coverage.',
        isInternal: true,
      },
    ],
  });

  console.log('✅ Created feedback comments');

  // Create announcements
  await prisma.announcement.createMany({
    data: [
      {
        title: 'Welcome to the HR APP',
        content: 'We are excited to launch our new HR feedback portal. Please share your thoughts!',
        category: 'COMPANY_NEWS',
        isPinned: true,
        isActive: true,
        createdBy: admin.id,
      },
      {
        title: 'New Health Benefits',
        content: 'We are adding dental and vision coverage to our health insurance plans starting next month.',
        category: 'BENEFIT',
        isPinned: false,
        isActive: true,
        createdBy: hrManager.id,
      },
      {
        title: 'Annual Performance Reviews',
        content: 'Performance review season is here. Please schedule your 1-on-1 with your manager.',
        category: 'HR_POLICY',
        isPinned: false,
        isActive: true,
        expiresAt: new Date(Date.now() + 30 * 24 * 60 * 60 * 1000), // 30 days from now
        createdBy: hrManager.id,
      },
      {
        title: 'Company Holiday Party',
        content: 'Join us for our annual holiday party on December 15th! RSVP by December 1st.',
        category: 'EVENT',
        isPinned: true,
        isActive: true,
        createdBy: admin.id,
      },
    ],
  });

  console.log('✅ Created 4 announcements');

  // Create notifications
  await prisma.notification.createMany({
    data: [
      {
        userId: employee1.id,
        type: 'ANNOUNCEMENT',
        title: 'New Announcement',
        message: 'Welcome to the HR APP',
        isRead: false,
        relatedEntityType: 'announcement',
        relatedEntityId: 1,
      },
      {
        userId: hrManager.id,
        type: 'FEEDBACK',
        title: 'New Feedback Submitted',
        message: 'Request for remote work policy',
        isRead: false,
        relatedEntityType: 'feedback',
        relatedEntityId: feedback2.id,
      },
      {
        userId: employee2.id,
        type: 'FEEDBACK',
        title: 'Feedback Status Updated',
        message: 'Your feedback status changed to UNDER_REVIEW',
        isRead: true,
        relatedEntityType: 'feedback',
        relatedEntityId: feedback2.id,
      },
    ],
  });

  console.log('✅ Created notifications');

  // Create sample direct conversation with messages
  const sampleParticipantKey = `${Math.min(employee1.id, employee2.id)}:${Math.max(employee1.id, employee2.id)}`;
  const directConversation = await prisma.directConversation.create({
    data: {
      participantKey: sampleParticipantKey,
      topic: 'Product launch sync',
      createdBy: employee1.id,
      participants: {
        create: [
          { userId: employee1.id },
          { userId: employee2.id },
        ],
      },
    },
  });

  const firstDirectMessage = await prisma.directMessage.create({
    data: {
      conversationId: directConversation.id,
      senderId: employee1.id,
      content: 'Hey Jane, wanted to sync up about the launch tasks.',
    },
  });

  const secondDirectMessage = await prisma.directMessage.create({
    data: {
      conversationId: directConversation.id,
      senderId: employee2.id,
      content: 'Sure thing! I can chat after lunch today.',
    },
  });

  await prisma.directConversation.update({
    where: { id: directConversation.id },
    data: {
      lastMessageAt: secondDirectMessage.createdAt,
    },
  });

  await prisma.directConversationParticipant.update({
    where: {
      conversationId_userId: {
        conversationId: directConversation.id,
        userId: employee1.id,
      },
    },
    data: {
      lastReadMessageId: secondDirectMessage.id,
    },
  });

  await prisma.directConversationParticipant.update({
    where: {
      conversationId_userId: {
        conversationId: directConversation.id,
        userId: employee2.id,
      },
    },
    data: {
      lastReadMessageId: firstDirectMessage.id,
    },
  });

  console.log('✅ Created sample direct conversation and messages');


  // Create sample birthday event for August
  const seedYear = new Date().getUTCFullYear();
  const birthdayEventMonth = 8;
  const birthdayEvent = await prisma.birthdayEvent.create({
    data: {
      year: seedYear,
      month: birthdayEventMonth,
      eventDate: new Date(Date.UTC(seedYear, birthdayEventMonth - 1, 25, 15, 0, 0)),
      title: 'August Birthday Celebration',
      description: 'Celebrate all August birthdays with cake and games!',
      location: 'Main Town Hall',
      createdById: hrManager.id,
    },
  });

  const augustBirthdayUsers = await prisma.user.findMany({
    where: {
      dateOfBirth: { not: null },
      isActive: true,
    },
  });

  const eligibleAugust = augustBirthdayUsers.filter(
    (user) => user.dateOfBirth && user.dateOfBirth.getUTCMonth() + 1 === birthdayEventMonth
  );

  for (const [index, user] of eligibleAugust.entries()) {
    await prisma.birthdayRegistration.create({
      data: {
        eventId: birthdayEvent.id,
        userId: user.id,
        rsvpStatus: index % 2 === 0 ? 'going' : 'pending',
        rsvpAt: index % 2 === 0 ? new Date() : null,
      },
    });
  }

  console.log('?? Created sample birthday event and registrations');


  // Create audit log entries
  await prisma.auditLog.createMany({
    data: [
      {
        userId: admin.id,
        action: 'CREATE_CHANNEL',
        entityType: 'channel',
        entityId: generalChannel.id,
        details: 'Created General Announcements channel',
        ipAddress: '127.0.0.1',
      },
      {
        userId: employee1.id,
        action: 'CREATE_FEEDBACK',
        entityType: 'feedback',
        entityId: feedback1.id,
        details: 'Submitted feedback: Improve office ergonomics',
        ipAddress: '127.0.0.1',
      },
    ],
  });

  console.log('✅ Created audit logs');

  console.log('');
  console.log('🎉 Database seeded successfully!');
  console.log('');
  console.log('📧 Demo Accounts (P@ssw0rd!):');
  console.log('  SUPERADMIN: sa@demo.local / P@ssw0rd!');
  console.log('  ADMIN:      admin@demo.local / P@ssw0rd!');
  console.log('  EMPLOYEE:   user@demo.local / P@ssw0rd!');
  console.log('');
  console.log('📧 Alternative Accounts (password123):');
  console.log('  ADMIN:      admin@company.com / password123');
  console.log('  HR:         hr@company.com / password123');
  console.log('  EMPLOYEE:   john.doe@company.com / password123');
  console.log('  EMPLOYEE:   jane.smith@company.com / password123');
  console.log('  EMPLOYEE:   bob.johnson@company.com / password123');
  console.log('  EMPLOYEE:   alice.williams@company.com / password123');
}

main()
  .catch((e) => {
    console.error('❌ Seed failed:', e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

File: backend/app/db/seed.py
"""
Database seed script - creates initial data.
Run this after migrations: python -m app.db.seed
"""
import sys
from pathlib import Path
from datetime import datetime, timedelta

# Add backend root to path
sys.path.insert(0, str(Path(__file__).parent.parent.parent))

from sqlalchemy import create_engine
from sqlalchemy.orm import sessionmaker

from app.core.config import settings
from app.core.security import hash_password
from app.db.base import Base
from app.modules.users.models import User, UserRole
from app.modules.channels.models import Channel, ChannelMember, MemberRole
from app.modules.feedback.models import Feedback, FeedbackComment, FeedbackStatus, FeedbackCategory
from app.modules.announcements.models import Announcement, AnnouncementCategory

# Import all models to ensure they're registered
from app.modules.users.models import User  # noqa: F401
from app.modules.channels.models import Channel, ChannelMember  # noqa: F401
from app.modules.feedback.models import Feedback, FeedbackComment  # noqa: F401
from app.modules.announcements.models import Announcement  # noqa: F401
from app.modules.notifications.models import Notification  # noqa: F401
from app.modules.files.models import File  # noqa: F401
from app.modules.admin.models import AuditLog  # noqa: F401


def seed_database():
    """Seed the database with initial data."""
    print("🌱 Seeding database...")

    # Create engine and session
    engine = create_engine(settings.DATABASE_URL)
    SessionLocal = sessionmaker(bind=engine)
    db = SessionLocal()

    try:
        # Create tables (in case migrations haven't run)
        Base.metadata.create_all(bind=engine)
        print("✅ Tables created/verified")

        # Check if superadmin exists
        existing_superadmin = (
            db.query(User).filter(User.email == "superadmin@company.com").first()
        )

        if existing_superadmin:
            print("ℹ️  Superadmin already exists")
        else:
            # Create superadmin
            superadmin = User(
                email="superadmin@company.com",
                password_hash=hash_password("Admin123!"),
                name="Super Admin",
                role=UserRole.SUPERADMIN,
                is_active=True,
            )
            db.add(superadmin)
            print("✅ Created superadmin user")

        # Create HR user
        existing_hr = db.query(User).filter(User.email == "hr@company.com").first()
        if not existing_hr:
            hr_user = User(
                email="hr@company.com",
                password_hash=hash_password("Hr123!"),
                name="HR Manager",
                department="Human Resources",
                role=UserRole.HR,
                is_active=True,
            )
            db.add(hr_user)
            print("✅ Created HR user")

        # Create employee user
        existing_employee = db.query(User).filter(User.email == "employee@company.com").first()
        if not existing_employee:
            employee = User(
                email="employee@company.com",
                password_hash=hash_password("Employee123!"),
                name="John Doe",
                department="Engineering",
                role=UserRole.EMPLOYEE,
                is_active=True,
            )
            db.add(employee)
            print("✅ Created employee user")
        
        # Create admin user
        existing_admin = db.query(User).filter(User.email == "admin@company.com").first()
        if not existing_admin:
            admin = User(
                email="admin@company.com",
                password_hash=hash_password("Admin123!"),
                name="Admin User",
                department="IT",
                role=UserRole.ADMIN,
                is_active=True,
            )
            db.add(admin)
            print("✅ Created admin user")
        
        # Create additional employees
        employees_data = [
            ("jane.smith@company.com", "Jane Smith", "Marketing"),
            ("mike.johnson@company.com", "Mike Johnson", "Sales"),
            ("sarah.wilson@company.com", "Sarah Wilson", "Engineering"),
        ]
        
        for email, name, dept in employees_data:
            existing = db.query(User).filter(User.email == email).first()
            if not existing:
                user = User(
                    email=email,
                    password_hash=hash_password("Employee123!"),
                    name=name,
                    department=dept,
                    role=UserRole.EMPLOYEE,
                    is_active=True,
                )
                db.add(user)
        print("✅ Created additional employee users")

        db.commit()
        
        # Get created users
        superadmin = db.query(User).filter(User.email == "superadmin@company.com").first()
        hr_user = db.query(User).filter(User.email == "hr@company.com").first()
        employee = db.query(User).filter(User.email == "employee@company.com").first()
        admin = db.query(User).filter(User.email == "admin@company.com").first()
        jane = db.query(User).filter(User.email == "jane.smith@company.com").first()
        mike = db.query(User).filter(User.email == "mike.johnson@company.com").first()
        sarah = db.query(User).filter(User.email == "sarah.wilson@company.com").first()
        
        all_users = [u for u in [superadmin, hr_user, employee, admin, jane, mike, sarah] if u]
        
        # Create default channels
        existing_general = db.query(Channel).filter(Channel.name == "general").first()
        if not existing_general:
            general_channel = Channel(
                name="general",
                description="General company announcements and discussions",
                is_public=True,
                created_by=hr_user.id,
            )
            db.add(general_channel)
            db.flush()  # Get the ID
            
            # Add all users as members
            for user in all_users:
                db.add(ChannelMember(
                    user_id=user.id,
                    channel_id=general_channel.id,
                    role=MemberRole.MEMBER,
                ))
            print("✅ Created 'general' channel with all users")
        
        existing_hr = db.query(Channel).filter(Channel.name == "hr-announcements").first()
        if not existing_hr:
            hr_channel = Channel(
                name="hr-announcements",
                description="HR policies, benefits, and important updates",
                is_public=True,
                created_by=hr_user.id,
            )
            db.add(hr_channel)
            db.flush()
            
            # Add HR as moderator
            db.add(ChannelMember(
                user_id=hr_user.id,
                channel_id=hr_channel.id,
                role=MemberRole.MODERATOR,
            ))
            # Add other users as members
            for user in [employee, jane, mike, sarah]:
                if user:
                    db.add(ChannelMember(
                        user_id=user.id,
                        channel_id=hr_channel.id,
                        role=MemberRole.MEMBER,
                    ))
            print("✅ Created 'hr-announcements' channel")
        
        # Create engineering channel
        existing_eng = db.query(Channel).filter(Channel.name == "engineering").first()
        if not existing_eng:
            eng_channel = Channel(
                name="engineering",
                description="Engineering team discussions",
                is_public=False,
                created_by=employee.id if employee else hr_user.id,
            )
            db.add(eng_channel)
            db.flush()
            
            # Add engineering team members
            for user in [employee, sarah]:
                if user:
                    db.add(ChannelMember(
                        user_id=user.id,
                        channel_id=eng_channel.id,
                        role=MemberRole.MEMBER,
                    ))
            print("✅ Created 'engineering' private channel")
        
        db.commit()
        
        # Create sample feedback
        general_channel = db.query(Channel).filter(Channel.name == "general").first()
        
        feedback_data = [
            {
                "title": "Office Temperature Too Cold",
                "content": "The air conditioning is set too cold in the main office. Many employees are complaining about the temperature.",
                "category": FeedbackCategory.WORKPLACE.value,
                "status": FeedbackStatus.PENDING.value,
                "submitted_by": employee.id if employee else None,
                "is_anonymous": False,
            },
            {
                "title": "Request for Remote Work Policy",
                "content": "Can we establish a clear remote work policy? Many employees would benefit from flexibility.",
                "category": FeedbackCategory.BENEFITS.value,
                "status": FeedbackStatus.REVIEWED.value,
                "submitted_by": jane.id if jane else None,
                "assigned_to": hr_user.id,
                "is_anonymous": False,
            },
            {
                "title": "Parking Space Issues",
                "content": "There aren't enough parking spaces for all employees. This has been an ongoing issue.",
                "category": FeedbackCategory.WORKPLACE.value,
                "status": FeedbackStatus.PENDING.value,
                "submitted_by": mike.id if mike else None,
                "is_anonymous": False,
            },
            {
                "title": "Anonymous Feedback About Team Communication",
                "content": "Some team members are not responding to emails in a timely manner, affecting project progress.",
                "category": FeedbackCategory.OTHER.value,
                "status": FeedbackStatus.RESOLVED.value,
                "submitted_by": sarah.id if sarah else None,
                "assigned_to": hr_user.id,
                "is_anonymous": True,
            },
        ]
        
        existing_feedback = db.query(Feedback).count()
        if existing_feedback == 0:
            for idx, feedback_info in enumerate(feedback_data, 1):
                feedback = Feedback(**feedback_info)
                db.add(feedback)
                db.flush()
                
                # Add a comment to some feedback
                if idx == 2:
                    comment = FeedbackComment(
                        feedback_id=feedback.id,
                        user_id=hr_user.id,
                        comment="We are reviewing parking options with building management.",
                    )
                    db.add(comment)
                elif idx == 4:
                    comment = FeedbackComment(
                        feedback_id=feedback.id,
                        user_id=hr_user.id,
                        comment="We've addressed this in the team meeting. Issue resolved.",
                    )
                    db.add(comment)
            
            print(f"✅ Created {len(feedback_data)} sample feedback items")
        
        db.commit()
        
        # Create sample announcements
        announcements_data = [
            {
                "title": "Welcome to the New HR Portal",
                "content": "We're excited to launch our new HR portal for employee feedback, announcements, and internal communication. Please explore the features and provide feedback!",
                "category": AnnouncementCategory.GENERAL,
                "is_pinned": True,
                "created_by": hr_user.id,
            },
            {
                "title": "Updated Remote Work Policy",
                "content": "Effective next month, employees can work remotely up to 2 days per week. Please coordinate with your manager and submit requests through the portal.",
                "category": AnnouncementCategory.POLICY,
                "is_pinned": True,
                "created_by": hr_user.id,
            },
            {
                "title": "Company Holiday Party - December 20th",
                "content": "Join us for our annual holiday celebration! Food, drinks, and entertainment. RSVP by December 10th.",
                "category": AnnouncementCategory.EVENT,
                "is_pinned": False,
                "created_by": hr_user.id,
                "expires_at": datetime.utcnow() + timedelta(days=30),
            },
            {
                "title": "New Health Insurance Benefits",
                "content": "We've expanded our health insurance options. Review the new plans and make your selections during open enrollment.",
                "category": AnnouncementCategory.BENEFIT,
                "is_pinned": False,
                "created_by": hr_user.id,
            },
        ]
        
        existing_announcements = db.query(Announcement).count()
        if existing_announcements == 0:
            for announcement_info in announcements_data:
                announcement = Announcement(**announcement_info)
                db.add(announcement)
            
            print(f"✅ Created {len(announcements_data)} sample announcements")
        
        db.commit()
        
        print("\n🎉 Database seeded successfully!")
        print("\n📋 Default Users:")
        print("   Superadmin: superadmin@company.com / Admin123!")
        print("   Admin:      admin@company.com / Admin123!")
        print("   HR:         hr@company.com / Hr123!")
        print("   Employee:   employee@company.com / Employee123!")
        print("   Jane:       jane.smith@company.com / Employee123!")
        print("   Mike:       mike.johnson@company.com / Employee123!")
        print("   Sarah:      sarah.wilson@company.com / Employee123!")
        print("\n📺 Default Channels:")
        print("   • general (public, 7 members)")
        print("   • hr-announcements (public, 5 members)")
        print("   • engineering (private, 2 members)")
        print("\n📝 Sample Data:")
        print(f"   • {len(feedback_data)} feedback items (various statuses)")
        print(f"   • {len(announcements_data)} announcements (2 pinned)")
        print(f"   • Feedback comments and assignments")

    except Exception as e:
        print(f"❌ Error seeding database: {e}")
        db.rollback()
        raise
    finally:
        db.close()


if __name__ == "__main__":
    seed_database()
